docs/tutorials/coord_sys_tutorial.md

```diff
@@ -0,0 +1,211 @@
+  |    |    |    |
+  |    |    |    | dx
+__|____|____|____|______\ x right
+  |    |    |    |      /
+  |    |    |    |
+  |    |____|____|
+  |         dy
+```
+
+Note that the box direction is always parallel with the edge <img src="https://render.githubusercontent.com/render/math?math=dx">.
+
+```
+y front
+  ^     _________
+ /|\   |    |    |
+  |    |    |    |
+  |    |    |    | dy
+  |    |____|____|____\  box direction (yaw=0)
+  |    |    |    |    /
+__|____|____|____|_________\ x right
+  |    |    |    |         /
+  |    |____|____|
+  |         dx
+  |
+```
+
+## Illustration of the three coordinate systems
+
+The illustration of the three coordinate systems is shown below:
+
+![coord_sys_all](../../resources/coord_sys_all.png)
+
+The three figures above are the 3D coordinate system while the three figures below are the bird's eye view.
+
+## Relation with raw coordinate systems of supported datasets
+
+### KITTI
+
+The raw annotation of KITTI is under Camera coordinate system, see [get_label_anno](../../tools/data_converter/kitti_data_utils.py). In MMDetection3D, to train models on KITTI, the data is first converted from camera to LiDAR, see [get_ann_info](../../mmdet3d/datasets/kitti_dataset.py).
+
+In SECOND, the LiDAR coordinate system for a box is defined as follows (a bird's eye view):
+
+![lidar](https://raw.githubusercontent.com/traveller59/second.pytorch/master/images/kittibox.png)
+
+
+
+For each box, the dimensions are <img src="https://render.githubusercontent.com/render/math?math=(w, l, h)">, and the reference direction for the yaw angle is the positive direction of the y axis. For more details, refer to the [repo](https://github.com/traveller59/second.pytorch#concepts).
+
+Our LiDAR coordinate system has two changes:
+
+- The yaw angle is defined to be right-handed instead of left-handed for consistency;
+- The box dimensions are <img src="https://render.githubusercontent.com/render/math?math=(l, w, h)"> instead of <img src="https://render.githubusercontent.com/render/math?math=(w, l, h)">, since <img src="https://render.githubusercontent.com/render/math?math=w"> corresponds to <img src="https://render.githubusercontent.com/render/math?math=dy"> and <img src="https://render.githubusercontent.com/render/math?math=l"> corresponds to <img src="https://render.githubusercontent.com/render/math?math=dx"> in KITTI.
+
+### Waymo
+
+We use the KITTI format data of Waymo dataset. Therefore, KITTI and Waymo also share the same coordinate system in our implementation.
+
+### NuScenes
+
+NuScenes provides a toolkit for evaluation, in which each box is wrapped into a `Box` instance. The coordinate system of `Box` is different from our LiDAR coordinate system in that the first two elements of the box dimension correspond to <img src="https://render.githubusercontent.com/render/math?math=(dy, dx)">, or <img src="https://render.githubusercontent.com/render/math?math=(w, l)">, respectively, instead of the reverse.
```

Comment:
Besides the exchange of `x` and `y`, NuscBox also has a different definition of rotation angle. If I remember correctly, it has yaw, pitch, roll. Its pitch is always (around) `pi/2`, roll always `0`, so we can only predict yaw. But when converting our box to NuscBox for evaluation, we need to manually rotate `pi/2` in pitch. See [here](https://github.com/open-mmlab/mmdetection3d/blob/78f456216f101b5b684db754176e0def17ab9d49/mmdet3d/datasets/nuscenes_mono_dataset.py#L672).

Should we also add this in the docs? Also, I am not sure why such pitch rotation is only performed in NuscMono dataset, while [Nusc3D](https://github.com/open-mmlab/mmdetection3d/blob/78f456216f101b5b684db754176e0def17ab9d49/mmdet3d/datasets/nuscenes_dataset.py#L596) dataset doesn't have such rotation. Is that because Mono box is in Camera coord while 3D box is in LiDAR coord?

Reply:
Hmmm, that's probably the reason. The subsequent transformation also has difference since one is in Cam coord while the other is in LiDAR coord (https://github.com/open-mmlab/mmdetection3d/blob/78f456216f101b5b684db754176e0def17ab9d49/mmdet3d/datasets/nuscenes_dataset.py#L613 v.s. https://github.com/open-mmlab/mmdetection3d/blob/78f456216f101b5b684db754176e0def17ab9d49/mmdet3d/datasets/nuscenes_mono_dataset.py#L686)

Maybe ask @Tai-Wang 's opinion on whether to add this to the coordinate system tutorial? This seems more like a nuScenes specific question.

Reply:
I need to spend some time looking into this more carefully.

Reply:
@Wuziyi616 @THU17cyz Yes, that problem only exists in the camera coordinate system. I have added related contents in the nuScenes dataset tutorial.

Reply:
Maybe here we can add a brief note and link to nuScenes tutorials.

Reply:
Sure